libs/cart/driver/magento/src/errors/transform.ts

```diff
@@ -1,7 +1,31 @@
+import { ApolloError } from 'apollo-client';
+
 import { daffTransformMagentoError } from '@daffodil/driver/magento';
+import { DaffCartDriverErrorMap } from '@daffodil/cart/driver';
+
+import { DaffCartMagentoErrorMap, DaffCartMagentoErrorMessageRegexMap } from './map';
+
+
+function transformMagentoCartGraphQlError(error: ApolloError): Error {
+	if (error.graphQLErrors.length) {
+    for (const code in DaffCartMagentoErrorMessageRegexMap) {
+      if (DaffCartMagentoErrorMessageRegexMap.hasOwnProperty(code)) {
+        const matchIndex = error.graphQLErrors[0].message.search(DaffCartMagentoErrorMessageRegexMap[code]);
+
+        if (matchIndex > -1 && DaffCartDriverErrorMap[code]) {
+          return new DaffCartDriverErrorMap[code](error.message)
+        }
+      }
+    }
+  }
 
-import { DaffCartMagentoErrorMap } from './map';
+  return daffTransformMagentoError(error, DaffCartMagentoErrorMap);
+};
 
 export function transformCartMagentoError(error) {
-  return daffTransformMagentoError(error, DaffCartMagentoErrorMap)
+  if (error.graphQLErrors) {
```

Comment:
I think you could just check error.graphQLErrors.length here and not need to check it in the `transformMagentoCartGraphQlError` function.

Reply:
thats better, changed